src/snowflake/cli/plugins/nativeapp/utils.py

```diff
@@ -82,3 +82,22 @@ def verify_no_directories(paths_to_sync: List[Path]):
 def verify_exists(path: Path):
     if not path.exists():
         raise ClickException(f"The following path does not exist: {path}")
+
+
+def get_all_file_paths_under_dir(directory: Path):
+    """
+    Recursively traverse the given directory and yield file paths. If the directory is a symlink, then the function follows the behavior of os.walk for symlinks.
+    The directory may be a relative path to the project root.
+    """
+    for root, _, files in os.walk(directory):
+        for file in files:
+            yield Path(os.path.join(root, file))
+
+
+def filter_files(generator: Generator, predicate_func: Callable):
```

Comment:
Is this utility necessary? Can it not be a generator comprehension? https://peps.python.org/pep-0289/

Reply:
Actually for this and the function above, I am not longer using these two. The original function that was using these two utilities has been replaced by Bruno's BundleMap implementation. So I have deleted these two utilities.